atlasdb-processors-tests/src/main/java/com/palantir/processors/TestInterfaceImpl.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.processors;
+
+@AutoDelegate(interfaceToExtend = TestInterface.class)
+public class TestInterfaceImpl implements AutoDelegate_TestInterface {
```

Comment:
Yep. I'm going to an approach that's going to generate abstract classes when extending a class and interfaces with default methods if the `typeToExtend` is an interface. There are not a lot of changes to the code, and it's the use case that Java's Proxy doesn't support (and we need it on our codebase), so I'm going to support it.